crates/gitbutler-reference/src/refname/virtual.rs

```diff
@@ -6,7 +6,9 @@ use super::error::Error;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct Refname {
-    // contains slug of the virtual branch name
+    /// contains slug of the virtual branch name.
+    /// Note that `Display` is implemented to put `refs/gitbutler/` in front of `branch`
+    /// so it's assume to be a valid branch name otherwise.
     pub branch: String,
```

Comment:
Just FYI, the entire `gitbutler-reference` crate can be considered on the "kill list". It's a bit of an incorrect abstraction and there are bunch of incorrect things that are modeled in it. I'd love to move away from it when we can, its just that it's used in a lot of places and it's removal is not straight forward.

(btw, this comment is not related to the docs update diff, i just needed a place to comment on the PR)

Reply:
Thanks for letting me know. Unless you think it's a problem, I'd keep changing code there as well, assuming that even if the crate should be dissolved, the code will just move to other places and/or be refactored.

In a way, this is what I think I will end up doing in small steps, even though I am less concerned about where code lives.